client/client_test.go

```diff
@@ -1594,6 +1597,93 @@ func testLocalSourceWithDiffer(t *testing.T, sb integration.Sandbox, d llb.DiffT
 	}
 }
 
+func testOCILayoutSource(t *testing.T, sb integration.Sandbox) {
+	requiresLinux(t)
+	c, err := New(context.TODO(), sb.Address())
+	require.NoError(t, err)
+	defer c.Close()
+
+	// create a tempdir where we will store the OCI layout
+	dir, err := os.MkdirTemp("", "buildkit-oci-layout")
+	require.NoError(t, err)
+	defer os.RemoveAll(dir)
+
+	// make an image that is exported there
+	busybox := llb.Image("busybox:latest")
+	st := llb.Scratch()
+
+	run := func(cmd string) {
+		st = busybox.Run(llb.Shlex(cmd), llb.Dir("/wd")).AddMount("/wd", st)
+	}
+
+	run(`sh -c "echo -n first > foo"`)
+	run(`sh -c "echo -n second > bar"`)
+
+	def, err := st.Marshal(sb.Context())
+	require.NoError(t, err)
+
+	outW := bytes.NewBuffer(nil)
+	attrs := map[string]string{}
+	_, err = c.Solve(sb.Context(), def, SolveOpt{
+		Exports: []ExportEntry{
+			{
+				Type:   ExporterOCI,
+				Attrs:  attrs,
+				Output: fixedWriteCloser(nopWriteCloser{outW}),
+			},
+		},
+	}, nil)
+	require.NoError(t, err)
+
+	// extract the tar stream to the directory as OCI layout
+	m, err := testutil.ReadTarToMap(outW.Bytes(), false)
+	require.NoError(t, err)
+
+	for filename, content := range m {
+		os.WriteFile(path.Join(dir, filename), content.Data, 0644)
+	}
+
+	var index ocispecs.Index
+	err = json.Unmarshal(m["index.json"].Data, &index)
+	require.NoError(t, err)
+	require.Equal(t, 1, len(index.Manifests))
+	digest := index.Manifests[0].Digest.Hex()
+
+	store, err := local.NewStore(dir)
+	require.NoError(t, err)
+
+	// reference the OCI Layout in a build
+	st = llb.OCILayout(dir, digest)
```

Comment:
better to not use var `dir` in here as there is no requirement that this needs to be a path. Just set it to a unique constant that is also used in `OCIStores` map.

Reply:
Could you also run a command on top of this state just to make sure that the marshaling works correctly

Reply:
> better to not use var dir in here as there is no requirement that this needs to be a path. Just set it to a unique constant that is also used in OCIStores map

Done, with comment added to explain it as well. I suspect there are other places where we actually used the dir, so that may break something, but CI should pick it up correctly.

Reply:
> Could you also run a command on top of this state just to make sure that the marshaling works correctly

I don't understand